cpp/src/parquet/page_index.cc

```diff
@@ -830,14 +894,13 @@ RowGroupIndexReadRange PageIndexReader::DeterminePageIndexRangesInRowGroup(
 // ----------------------------------------------------------------------
 // Public factory functions
 
-std::unique_ptr<ColumnIndex> ColumnIndex::Make(const ColumnDescriptor& descr,
-                                               const void* serialized_index,
-                                               uint32_t index_len,
-                                               const ReaderProperties& properties) {
+std::unique_ptr<ColumnIndex> ColumnIndex::Make(
+    const ColumnDescriptor& descr, const void* serialized_index, uint32_t index_len,
+    const ReaderProperties& properties, const std::shared_ptr<Decryptor>& decryptor) {
   format::ColumnIndex column_index;
   ThriftDeserializer deserializer(properties);
   deserializer.DeserializeMessage(reinterpret_cast<const uint8_t*>(serialized_index),
```

Comment:
Can `DeserializeMessage` be updated to take a raw `Decryptor*`? It doesn't need to take ownership...

Reply:
Same for `SerializeMessage`, incidentally.

Reply:
It seems to involve a lot of change. I will try to take a deep look.